chroma_core/models/stratagem.py

```diff
@@ -0,0 +1,214 @@
+
+    def description(self):
+        return help_text["configure_stratagem_description"]
+
+    def get_steps(self):
+        steps = [(ConfigureStratagemTimerStep, {})]
+
+        return steps
+
+
+class RunStratagemStep(Step):
+    def run(self, args):
+        host = args["host"]
+        path = args["path"]
+
+        def _get_body(mount_point):
+            return {
+                "dump_flist": False,
+                "device": {"path": path, "groups": ["size_distribution", "warn_purge_times"]},
+                "groups": [
+                    {
+                        "rules": [
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": "< size 1048576",
+                                "argument": "smaller_than_1M",
+                            },
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": "&& >= size 1048576 < size 1048576000",
+                                "argument": "not_smaller_than_1M_and_smaller_than_1G",
+                            },
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": ">= size 1048576000",
+                                "argument": "not_smaller_than_1G",
+                            },
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": ">= size 1048576000000",
+                                "argument": "not_smaller_than_1T",
+                            },
+                        ],
+                        "name": "size_distribution",
+                    },
+                    {
+                        "rules": [
+                            {
+                                "action": "LAT_SHELL_CMD_FID",
+                                "expression": "< atime - sys_time 18000000",
+                                "argument": "fids_expiring_soon",
+                            },
+                            {
+                                "action": "LAT_SHELL_CMD_FID",
+                                "expression": "< atime - sys_time 5184000000",
+                                "argument": "fids_expired",
+                            },
+                        ],
+                        "name": "warn_purge_times",
+                    },
+                ],
+            }
+
+        body = _get_body(path)
+        result = self.invoke_rust_agent(host, "start_scan_stratagem", body)
+        result = json.loads(result)
+
+        if "Err" in result:
+            self.log("Error Scanning {}: \n{}".format(path, json.dumps(result["Err"], indent=2)))
+        else:
+            self.log("Successfully scanned {}:\n{}".format(path, json.dumps(result["Ok"], indent=2)))
+
+
+class RunStratagemJob(Job):
+    mdt_id = models.IntegerField()
```

Comment:
Again, this will need to be bounded per-filesystem.

We don't want to scan one filesystem and then try to use it's results on another.

Reply:
I'll handle this by passing in the filesystem id and only retrieve mdt's associated with that filesystem.

Reply:
Sounds good. In addition, you need to make sure the servers the MDTs reside on have the proper profile installed.

Reply:
👍